### PR TITLE
Add service accounts to `prowgen` tool

### PR DESCRIPTION
**Pull request by upodroid**
/cc @chaodaiG @howardjohn 

Hi, we are using this generator at knative/test-infra and I need urgent support for specifying service accounts. This change will allow us to safely add all PodSpec parameters except Container, Volumes.

### CODE REVIEW

**Comment by chaodaiG** [comment]
/cc @chizhg 
who has more context on this
/uncc
please add me back if my approval is needed

File: tools/prowgen/pkg/spec/spec.go [hunk 1]
```diff
@@ -143,6 +143,7 @@ type RequirementPreset struct {
 	Volumes      []v1.Volume       `json:"volumes,omitempty"`
 	VolumeMounts []v1.VolumeMount  `json:"volumeMounts,omitempty"`
 	Args         []string          `json:"args,omitempty"`
+	Spec         *v1.PodSpec       `json:"spec,omitempty"`
```
Comment (chizhg) [inline tools/prowgen/pkg/spec/spec.go]: Nit: I think it's better to name the json key as `podSpec`, and add a comment to explain when to use it - when you want to configure presets for fields in `v1.PodSpec` other than the `metadata` and `containers`.

**Review by chizhg — APPROVED** [review]

File: tools/prowgen/pkg/spec/spec.go (hunk 1 above)
Comment (upodroid) [reply tools/prowgen/pkg/spec/spec.go]: Fixed

File: tools/prowgen/pkg/decorator/requirement.go [hunk 2]
```diff
@@ -66,13 +67,13 @@ func ApplyRequirements(job *config.JobBase, requirements, excludedRequirements [
...
+			mergeRequirement(annotations, labels, spec, spec.Containers, &spec.Volumes, req)
 		}
 	}
 }
 
-// mergeRequirement will overlay the requirement on the existing job spec.
-func mergeRequirement(annotations, labels map[string]string, containers []v1.Container, volumes *[]v1.Volume,
+// mergeRequirement will overlay the requirement on the existing job spec. Use mergo for all keys except containers and metadata
+func mergeRequirement(annotations, labels map[string]string, spec *v1.PodSpec, containers []v1.Container, volumes *[]v1.Volume,
 	req spec.RequirementPreset) {
```
Comment (ericvn) [inline tools/prowgen/pkg/decorator/requirement.go:77]: Just pass `spec` and get containers from that?
```suggestion
func mergeRequirement(annotations, labels map[string]string, spec *v1.PodSpec, volumes *[]v1.Volume,
	req spec.RequirementPreset) {
	containers := spec.Containers
```

Comment (upodroid) [reply tools/prowgen/pkg/decorator/requirement.go:77]: That won't be helpful. The containers are processed by another function `createContainer` in `generate.go`. I would need to do a rewrite and modify the istio prow configuration.

**Comment by upodroid** [comment]
Can I get this approved and merged please? Thanks

**Review by ericvn — APPROVED** [review]

**Pull request merged**